### PR TITLE
Add `resetServerContext` function for SSR

### LINKED ISSUE #323: Way to reset data-react-beautiful-dnd-draggable for server side rendering
## Bug or feature request?
Bug

### Expected behavior
`data-react-beautiful-dnd-draggable` shouldn't increment based on how many times it was run on the server but start fresh for each request.

### Actual behavior
Each time the server render is hit `data-react-beautiful-dnd-draggable` is incremented by 1.
i.e. after 6 reloads console throws:
```
Warning: Prop `data-react-beautiful-dnd-draggable` did not match. Server: "6" Client: "0"
```

### Steps to reproduce
Server side render and reload couple times, each reload the `data-react-beautiful-dnd-draggable` gets incremented by 1.

### Browser version
Chrome

https://github.com/atlassian/react-beautiful-dnd/blob/master/src/view/style-marshal/style-marshal.js#L11

Possible solution to this would be to expose a function that resets the counter that one could call during a server side render like so:
```
export function resetCounter() {
    count = 0;
}
```

### PR DESCRIPTION
**Pull request by Blasz**
Closes #323.

I've abstracted the reset method solution from #337 into separate server + style context functions to make it a little bit more future proof.

I've added both integration and unit tests for the new functionality and fixed the context from being shared between unit tests by calling these functions in `beforeEach`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1250,6 +1250,22 @@ It is possible for your `Draggable` to contain interactive elements. By default
 
 You can opt out of this behavior by adding the `disableInteractiveElementBlocking` prop to a `Draggable`. However, it is questionable as to whether you should be doing so because it will render the interactive element unusable. If you need to *conditionally* block dragging from interactive elements you can add the `disableInteractiveElementBlocking` prop to opt out of the default blocking and monkey patch the `dragHandleProps (DragHandleProps)` event handlers to disable dragging as required.
 
+## `resetServerContext`
```
Comment (alexreardon) [inline README.md]: Do you think we should make a new section 'Server side rendering' with `resetServerContext` underneath?

Comment (Blasz) [reply README.md]: I was considering this as well but ultimately opted to keep the heading as the API method to be consistent with our other API exports. I think either way is fine though, up to you.

Comment (alexreardon) [reply README.md]: Agreed - let's leave it

**Comment by alexreardon** [comment]
pingu2

**Pull request merged**